src/Microsoft.DotNet.Build.Tasks/ZipFileCreateFromDependencyLists.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class ZipFileCreateFromDependencyLists : Task
+    {
+        /// <summary>
+        /// List of dependency text files to be unified.
+        /// </summary>
+        [Required]
+        public string DependencyListsFolder { get; set; }
+
+        /// <summary>
+        /// The path of the archive to be created.
+        /// </summary>
+        [Required]
+        public string DestinationArchive { get; set; }
+
+        [Required]
+        /// <summary>
+        /// Base path for paths specified in txt files in DependencyListsFolder
+        /// </summary>
+        public string RelativePathBaseDirectory { get; set; }
+
+        /// <summary>
+        /// Indicates if the destination archive should be overwritten if it already exists.
+        /// </summary>
+        public bool OverwriteDestination { get; set; } = false;
+
+        public override bool Execute()
+        {
+            Log.LogMessage($"DependencyListsFolder = {DependencyListsFolder}");
+            Log.LogMessage($"DestinationArchive = {DestinationArchive}");
+            Log.LogMessage($"RelativePathBaseDirectory = {RelativePathBaseDirectory}");
+            Log.LogMessage($"OverwriteDestination = {OverwriteDestination}");
+
+            try
+            {
+                if (File.Exists(DestinationArchive))
+                {
+                    if (OverwriteDestination == true)
+                    {
+                        Log.LogMessage(MessageImportance.Low, $"{DestinationArchive} already existed, deleting before zipping...");
+                        File.Delete(DestinationArchive);
+                    }
+                    else
+                    {
+                        Log.LogWarning($"'{DestinationArchive}' already exists, no change will occur. Did you forget to set '{nameof(OverwriteDestination)}' to true?");
+                    }
+                }
+
+                Log.LogMessage(MessageImportance.High, $"Compressing files listed in {DependencyListsFolder} into {DestinationArchive}...");
+                if (!Directory.Exists(Path.GetDirectoryName(DestinationArchive)))
+                {
+                    Directory.CreateDirectory(Path.GetDirectoryName(DestinationArchive));
+                }
+                List<string> filesForArchiving = GetUniquePaths(DependencyListsFolder, "*");
```

Comment:
We should probably make the file pattern `*.dependencylist.txt` just to be sure we don't try reading other files. Actually thinking about it a little more I think I would build this list of files in the targets file and pass them to the task, so that knowledge isn't hardcoded in the task.